### PR TITLE
Configurable favicon

### PR DESCRIPTION
**Pull request by Reigel**
Add a new optional entry in config file favicon.
Use default favicon if not configured.

### CODE REVIEW

File: kansha/app/view.py [hunk 1]
```diff
@@ -178,8 +178,8 @@ def render(self, h, comp, *args):
 
 
 @presentation.render_for(App)
-def render(self, h, comp, *args):
-    favicon_url = h.head.static_url + "img/favicon.ico"
+def render_app(self, h, comp, *args):
+    favicon_url = self.favicon if self.favicon else h.head.static_url + "img/favicon.ico"
```
Comment (ygravrand) [inline kansha/app/view.py]: suggestion:
```favicon_url = self.favicon or h.head.static_url + "img/favicon.ico"```

File: doc/configuration.rst [hunk 2]
```diff
@@ -144,6 +145,9 @@ banner
 custom_css
     Path to a CSS file that will be applied to every page, after the default styles, so you can amend and personalize the look of your site.
 
+custom_favicon
+    Path to a favicon file that will be applied to your site.
```
Comment (ygravrand) [inline doc/configuration.rst]: Suggestion: Path (relative to ``static`` directory) to a favicon...

File: conf/kansha.cfg [hunk 3]
```diff
@@ -7,6 +7,7 @@ as_root = on
 title = Kansha
 banner = "Open Source Collaborative Tool"
 custom_css =
+custom_favicon =
```
Comment (nagareproject) [inline conf/kansha.cfg]: No always defined 'custom_favicon' but optional 'favicon'

File: doc/configuration.rst [hunk 4]
```diff
@@ -27,6 +27,7 @@ Kansha features can be activated and customized with a configuration file like t
     title = <<APP_TITLE>> # should be short!
     banner = <<LONG_TITLE>> # or motto/slogan or empty
     custom_css = <<CUSTOM_CSS>>  # path or empty
+    custom_favicon = <<CUSTOM_FAVICON>>  # path or empty
```
Comment (nagareproject) [inline doc/configuration.rst]: favicon = <<CUSTOM_FAVICON>>  # path

File: doc/configuration.rst (hunk 2 above)
Comment (nagareproject) [inline doc/configuration.rst]: +favicon
+    Path to a favicon file that will be applied to your site.

File: kansha/app/app.py [hunk 5]
```diff
@@ -212,6 +212,7 @@ def __init__(self, app_title, custom_css, mail_sender, cfg, assets_manager,
 
         self.app_title = app_title
         self.app_banner = cfg['pub_cfg']['banner']
+        self.favicon = cfg['pub_cfg']['custom_favicon']
```
Comment (nagareproject) [inline kansha/app/app.py]: self.favicon = cfg['pub_cfg']['favicon']

File: kansha/app/app.py [hunk 6]
```diff
@@ -239,6 +240,7 @@ class WSGIApp(wsgi.WSGIApp):
                         'title': 'string(default="")',
                         'banner': 'string(default="")',
                         'custom_css': 'string(default="")',
+                        'custom_favicon': 'string(default="")',
```
Comment (nagareproject) [inline kansha/app/app.py]: 'favicon': 'string(default="img/favicon.ico")',

File: kansha/app/app.py [hunk 7]
```diff
@@ -314,7 +316,8 @@ def set_config(self, config_filename, conf, error):
         tpl_cfg = conf['application']['templates']
         pub_cfg = {
             'disclaimer': conf['application']['disclaimer'].decode('utf-8'),
-            'banner': conf['application']['banner'].decode('utf-8')
+            'banner': conf['application']['banner'].decode('utf-8'),
+            'custom_favicon': conf['application']['custom_favicon'].decode('utf-8')
```
Comment (nagareproject) [inline kansha/app/app.py]: +            'favicon': conf['application']['favicon'].decode('utf-8')

File: kansha/app/view.py (hunk 1 above)
Comment (nagareproject) [inline kansha/app/view.py]: favicon_url = absolute_url(self.favicon, h.head.static_url)

(from nagare.namespaces.xhtml import absolute_url)

**Pull request merged**